instructions.md

```diff
@@ -0,0 +1,140 @@
+
+* For provisioning, we connect `em1` to the bmi provisioning network natively.
+
+* To connect your node to internet, connect any of your nics to the `internet` network
+either natively or as a tagged network. Making it tagged would require you to
+do some additional configuration on your node. Recommended way to do is to connect em2 to `internet` natively.
+If you know what you are doing, you can ignore this section.
+
+* If you require a public IP address for your node, talk to Naved/Rado.
+
+## Using HIL
+
+1. Logon to the kumo-hil-client (available over Internet).
+  `ssh username@kumo-hil.massopen.cloud`
+
+2. Export your HIL credentials that an admin should have given you.
+
+```
+export HIL_USERNAME=username
+export HIL_PASSWORD=password
+export HIL_ENDPOINT='http://192.168.100.210:80'
+```
+
+You could put these in your bashrc so they are automatically in your environment
+when you login.
+
+3. Basic HIL commands:
+
+* `hil list_project_nodes <project-name>`
+    This will list all the nodes that your project has access to.
+
+* `hil show_node <node-name>`
+    This will show you the information about your node.
+
+* `hil node_connect_network <node-name> <nic-name> <network-name> <channel>`
+    to connect your node's nic to a network. Channel could either be `vlan/native`
+    or `vlan/<vlan-id>`.
+
+* to find the vlan-id of a network, run `hil show_network <network-name>`
+
+* `hil node_detach_network <node-name> <nic-name> <network-name>`
+    this will remove <network-name> from your node's nic.
+
+* `hil node_power_cycle <node-name>`.
+    to power cycle a node.
+
+* `hil node_power_off <node-name>`.
+    to power off a node.
+
+
+## Using BMI
+
+1. Logon to the BMI machine (available only through the HIL client).
+    `ssh username@kumo-bmi-no-seccloud.infra.massopen.cloud`
+    or use it's ip address which is 192.168.100.142
+
+2. Once logged in, export your HIL credentials explained above. In addition to that
+do this:
+    `export BMI_CONFIG=/etc/bmi/bmiconfig.cfg`
+
+Test bmi works by running `bmi db ls` command.
+
+
+3. Run this command to see a list of available images. The project name is the same name as in HIL.
+    `bmi ls <project-name>`
+
+
+4. If this is the first time you got a node, and you want to provision it with an OS using BMI.
+Run this command:
+    `bmi provision <project-name> <node-name> <image-name> bmi-provision-net-no-seccloud em1`
+
+
+    * Here bmi-provision-net-no-seccloud is the name of the provisioning network,
+    and em1 is the name of the nic. You could change it if you know what you are doing.
+
+    * Make sure that the network `bmi-provision-net-no-seccloud` is not already connected.
+    If it is, then disconnect that network using HIL and then run the BMI command again.
+
+    * The command should return `success` within 5-10 seconds.
+
+
+If you had already provisioned a node using bmi, and just want to reconnect your image
+to your node, run this script available in your path.
```

Comment:
tell them where they can find the script.

Reply:
You have to talk about the config file too which they need to put BMI server IP address in it.

Reply:
I think we should put the script in `/usr/local/bin/` that way it will be available globally.

Reply:
That is a good idea. or even in /home will suffice.

Reply:
wait, it should not be on the node itself. Users are going to run it from their managing machine.

Reply:
Not sure what you are saying.

This will be run from the bmi vm. so on the bmi vm, we put the file in `/usr/local/bin/` that way users can do `reconnect.sh <project> <node>` and they don't have to specify a path to it now since /usr/local/bin is in path by default.